test/e2e/kubectl/rollout.go

```diff
@@ -0,0 +1,145 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// OWNER = sig/cli
+
+package kubectl
+
+import (
+	"context"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	"github.com/onsi/ginkgo/v2"
+
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	commonutils "k8s.io/kubernetes/test/e2e/common"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2edeployment "k8s.io/kubernetes/test/e2e/framework/deployment"
+	e2ekubectl "k8s.io/kubernetes/test/e2e/framework/kubectl"
+	admissionapi "k8s.io/pod-security-admission/api"
+)
+
+const (
+	labeledDeploymentFilename = "labeled-deployment.yaml.in"
+)
+
+var _ = SIGDescribe("Kubectl rollout", func() {
+	defer ginkgo.GinkgoRecover()
+	f := framework.NewDefaultFramework("kubectl-rollout")
+	f.NamespacePodSecurityLevel = admissionapi.LevelBaseline
+
+	var c clientset.Interface
+	var ns string
+	ginkgo.BeforeEach(func() {
+		c = f.ClientSet
+		ns = f.Namespace.Name
+	})
+
+	ginkgo.Describe("undo", func() {
+		deploymentYaml := commonutils.SubstituteImageName(string(readTestFileOrDie(labeledDeploymentFilename)))
```

Comment:
We want to  test that the deployment has only the original set of labels and annotations? To confirm this, it might be good to add some to the original yaml. And I'm not sure whether changing the  existing files will break sth, so I created a new one. If it's ok I can add some labels to the existing file and then use it :)

Reply:
kindly ping @soltysh :)

Reply:
Right, but you only inject new envs, thus re-using all the labels and annotations just to ensure they don't change after rollback, so what's stopping you from using an existing deployment file?

Reply:
Well, it's not that complicated. I'm just not sure if it's ok to modify an existing file.  Updated to use `httpd yaml` now. I'll squash if the rest is ok.

Reply:
ping @soltysh